test/unit/io/sam_file/format_sam_test.cpp

```diff
@@ -125,64 +125,224 @@ struct sam_format : public sam_file_data
 // since BAM uses the same read header function from SAM, it only needs to be tested once
 TEST_F(sam_format, header_errors)
 {
+
+    // unsupported @-tag tests
     {
         std::string header_str
         {
-            "@HD\tVN:1.0\tTT:this is not a valid tag\n"
+            "@HA\tthis is not a valid tag\n"
         };
         std::istringstream istream(header_str);
         seqan3::sam_file_input fin{istream, seqan3::format_sam{}};
-        EXPECT_THROW(fin.begin(), seqan3::format_error);
+        EXPECT_THROW(fin.begin(), seqan3::format_error);    // "Illegal SAM header tag: @HA"
```

Comment:
```suggestion
        EXPECT_THROW(fin.begin(), seqan3::format_error);
```

Until we actually test the error messages (which we currently cannot), I wouldn't add the expected error message as they might change and it's quite clear what you are testing (`// unsupported @-tag tests` + `@HA this is not a valid tag`)